### PR TITLE
Determine query version for indexedb cache automatically.

### PR DESCRIPTION
**Pull request by salazarm**
## Summary & Motivation

Instead of requiring devs to manually update a version number, we instead hash the query to determine the version number.

The `print` usage doesn't seem to be expensive.

## How I Tested These Changes

Using local-host-cloud I loaded the page after changing a hash and saw that we refetch the LocationWorkspaceQuery and on reload we don't since the hash it the same. I then made a change to the query (removed a field) and reloaded the page and saw that we correctly fetched LocationWorkspaceQuery again.

## Changelog

NOCHANGELOG

### CODE REVIEW

File: js_modules/dagster-ui/packages/ui-core/src/search/useIndexedDBCachedQuery.tsx [hunk 1]
```diff
@@ -14,11 +16,16 @@ import {CompletionType, useBlockTraceUntilTrue} from '../performance/TraceContex
...
-  version: number;
+  version: number | string;
 };
 
 export const KEY_PREFIX = 'indexdbQueryCache:';
 
+export function getQueryVersion(query: DocumentNode) {
+  const printedQuery = print(query);
+  return sha256.create().update(printedQuery).hex();
+}
```
Comment (alangenfeld) [inline js_modules/dagster-ui/packages/ui-core/src/search/useIndexedDBCachedQuery.tsx:27]: how often do we end up running this? i see its gated in useMemo

**Review by alangenfeld — COMMENTED** [review]
can we generate the version string at graphql codegen time? That would be ideal since we already require running that generate step on graphql query changes

**Comment by salazarm** [comment]
> can we generate the version string at graphql codegen time? That would be ideal since we already require running that generate step on graphql query changes

The issue is that the query needs to be evaluated with all of its fragments. So I'd need to run the code to cache it at compile time. Probably doable but I don't know a quick and easy way to do it.

**Comment by salazarm** [comment]
@alangenfeld I'm familiar with our graphql codegen setup. It doesn't actually evaluate any javascript. It's all just static AST parsing/transformations.

**Comment by salazarm** [comment]
Hmm actually, maybe I can use the generate types to create the hash... I'll look into this

**Comment by salazarm** [comment]
I think I have an approach by letting graphql-codegen-persisted-query-ids generate the client.json file with hashes and then a script to move those hashes to the type files and export them so that we end up with files looking like this:

```
// Generated GraphQL types, do not edit manually.

import * as Types from '../../../graphql/types';

export type CodeLocationStatusQueryVariables = Types.Exact<{ [key: string]: never }>;

export type CodeLocationStatusQuery = {
  __typename: 'Query';
  locationStatusesOrError:
    | { __typename: 'PythonError' }
    | {
        __typename: 'WorkspaceLocationStatusEntries';
        entries: Array<{
          __typename: 'WorkspaceLocationStatusEntry';
          id: string;
          name: string;
          loadStatus: Types.RepositoryLocationLoadStatus;
          updateTimestamp: number;
        }>;
      };
};

// Version of this query
export const CodeLocationStatusQueryVersion = 'd2d2c4f2e7e3a4d3b...'; // Has
```

**Pull request closed without merging**